audio/audio_hw.c

```diff
@@ -396,9 +396,82 @@ static int out_set_parameters(struct audio_stream *stream, const char *kvpairs)
     return 0;
 }
 
-static char * out_get_parameters(const struct audio_stream *stream, const char *keys)
+static void get_audio_format(struct str_parms *str_parms,
+    audio_format_t audio_format)
 {
-    return strdup("");
+    const char *format;
+
+    switch (audio_format) {
+    case AUDIO_FORMAT_PCM:
+        format = "AUDIO_FORMAT_PCM";
+        break;
+    case AUDIO_FORMAT_MP3:
+        format = "AUDIO_FORMAT_MP3";
+        break;
+    case AUDIO_FORMAT_AMR_NB:
+        format = "AUDIO_FORMAT_AMR_NB";
+        break;
+    case AUDIO_FORMAT_AMR_WB:
+        format = "AUDIO_FORMAT_AMR_WB";
+        break;
+    case AUDIO_FORMAT_AAC:
+        format = "AUDIO_FORMAT_AAC";
+        break;
+    case AUDIO_FORMAT_HE_AAC_V1:
+        format = "AUDIO_FORMAT_HE_AAC_V1";
+        break;
+    case AUDIO_FORMAT_HE_AAC_V2:
+        format = "AUDIO_FORMAT_HE_AAC_V2";
+        break;
+    case AUDIO_FORMAT_VORBIS:
+        format = "AUDIO_FORMAT_VORBIS";
+        break;
+    case AUDIO_FORMAT_MAIN_MASK:
+        format = "AUDIO_FORMAT_MAIN_MASK";
+        break;
+    case AUDIO_FORMAT_SUB_MASK:
+        format = "AUDIO_FORMAT_SUB_MASK";
+        break;
```

Comment:
Delete these. AUDIO_FORMAT_MAIN_MASK and AUDIO_FORMAT_SUB_MASK are not audio formats. They are bitmasks for logical-AND with the format value.

Reply:
Done.